### PR TITLE
Fix post-processing of conflicts and add new test

### PR DESCRIPTION
**Pull request by pazembrz**
Additionally few tests were fixed as they was not checking what they should check.

### CODE REVIEW

File: tests/unit/test_api.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def test_merger_handles_list_deletions():
     ]
 
     merged, conflict = merge(root, head, update, head_source='arxiv')
-    assert merged == expected_merged
+    assert sorted(merged.items()) == sorted(expected_merged.items())
```
Comment (michamos) [inline tests/unit/test_api.py:308]: isn't this basically a less efficient way to do the same thing? two dicts are equal iff they have the same keys and the two values for a given key are equal.

File: tests/unit/test_api.py [hunk 2]
```diff
@@ -362,20 +362,24 @@ def test_merger_handles_authors_with_correct_ordering():
     ]}
 
     merged, conflict = merge(root, head, update, head_source='arxiv')
-    assert merged == expected_merged
+    assert sorted(merged.items()) == sorted(expected_merged.items())
```
Comment (michamos) [inline tests/unit/test_api.py:365]: same

**Review by michamos — APPROVED** [review]
Looks good :+1: minor comment about tests

File: tests/unit/test_api.py (hunk 1 above)
Comment (pazembrz) [reply tests/unit/test_api.py]: Yes, you are right

File: tests/unit/test_api.py (hunk 2 above)
Comment (pazembrz) [reply tests/unit/test_api.py]: :+1:

File: inspire_json_merger/pre_filters.py [hunk 3]
```diff
@@ -62,6 +62,11 @@ def keep_only_update_source_in_field(field, root, head, update):
             ``root`` and ``head``.
     """
     update_sources = {source.lower() for source in get_value(thaw(update), '.'.join([field, 'source']), [])}
```
Comment (michamos) [inline inspire_json_merger/pre_filters.py:64]: maybe store `thaw(update)` so you don't need to compute it twice?

**Review by michamos — APPROVED** [review]
small comment, but :+1: for the rest

File: inspire_json_merger/pre_filters.py (hunk 3 above)
Comment (pazembrz) [reply inspire_json_merger/pre_filters.py:64]: :+1:

**Pull request merged**